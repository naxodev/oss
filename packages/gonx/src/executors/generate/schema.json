{
  "version": 2,
  "outputCapture": "direct-nodejs",
  "$schema": "https://json-schema.org/schema",
  "title": "Generate executor",
  "description": "Runs code generation using the `go generate` command",
  "type": "object",
  "properties": {
    "env": {
      "type": "object",
      "additionalProperties": {
        "type": "string"
      },
      "description": "Environment variables to set when running the executor"
    },
    "flags": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Flags to pass to the go generate command"
    }
  }
}
