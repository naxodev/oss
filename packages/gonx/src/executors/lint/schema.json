{
  "$schema": "https://json-schema.org/schema",
  "version": 2,
  "title": "Lint executor",
  "description": "Formats and lints a Go project",
  "type": "object",
  "properties": {
    "linter": {
      "type": "string",
      "description": "The command to execute when running the lint executor instead of go fmt.",
      "default": "go fmt"
    },
    "args": {
      "type": "array",
      "description": "Extra args when linting the project",
      "items": {
        "type": "string"
      }
    }
  },
  "required": []
}
