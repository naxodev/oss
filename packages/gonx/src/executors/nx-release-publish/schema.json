{
  "$schema": "https://json-schema.org/schema",
  "version": 2,
  "title": "NxReleasePublish executor",
  "description": "Publish a Go module to a registry as part of the release process",
  "type": "object",
  "properties": {
    "moduleRoot": {
      "type": "string",
      "description": "Path to the directory containing the go.mod file. Defaults to the project root."
    },
    "dryRun": {
      "type": "boolean",
      "description": "Whether to perform a dry run (no actual publishing)",
      "default": false
    }
  },
  "required": []
}
