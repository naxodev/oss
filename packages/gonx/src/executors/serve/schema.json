{
  "$schema": "https://json-schema.org/schema",
  "version": 2,
  "title": "Serve executor",
  "description": "Run a Go application",
  "type": "object",
  "properties": {
    "main": {
      "type": "string",
      "description": "Path to the file containing the main() function"
    },
    "cmd": {
      "type": "string",
      "description": "Name of the go binary to use",
      "default": "go"
    },
    "cwd": {
      "type": "string",
      "description": "Working directory from which to run the application"
    },
    "args": {
      "type": "array",
      "description": "Extra args when starting the app",
      "items": {
        "type": "string"
      }
    },
    "env": {
      "type": "object",
      "description": "Environment variables to set when running the application"
    }
  },
  "required": ["main"]
}
