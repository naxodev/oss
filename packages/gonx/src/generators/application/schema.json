{
  "$schema": "https://json-schema.org/schema",
  "$id": "@naxodev/gonx:application",
  "title": "Go application generator options",
  "type": "object",
  "properties": {
    "directory": {
      "description": "The directory of the new application.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Which directory do you want to create the application in?"
    },
    "name": {
      "description": "The name of the application.",
      "type": "string",
      "pattern": "^[a-zA-Z][^:]*$",
      "x-priority": "important"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the application (used for linting)."
    },
    "skipFormat": {
      "type": "boolean",
      "description": "Skip formatting files.",
      "default": false,
      "x-priority": "internal"
    }
  },
  "required": ["directory"]
}
