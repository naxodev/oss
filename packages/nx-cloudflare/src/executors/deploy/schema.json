{
  "$schema": "https://json-schema.org/schema",
  "version": 2,
  "title": "Deploy executor",
  "description": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Name of the Worker."
    },
    "noBundle": {
      "type": "boolean",
      "default": false,
      "description": "Skip Wranglerâ€™s build steps and directly deploy script without modification. Particularly useful when using custom builds."
    },
    "env": {
      "type": "string",
      "description": "Perform on a specific environment."
    },
    "outdir": {
      "type": "string",
      "description": "Path to directory where Wrangler will write the bundled Worker files."
    },
    "compatibilityDate": {
      "type": "string",
      "description": "A date in the form yyyy-mm-dd, which will be used to determine which version of the Workers runtime is used."
    },
    "compatibilityFlags": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Flags to use for compatibility checks."
    },
    "latest": {
      "type": "boolean",
      "default": true,
      "description": "Use the latest version of the Workers runtime."
    },
    "assets": {
      "type": "string",
      "description": "Root folder of static assets to be served. Unlike --site, --assets does not require a Worker script to serve your assets."
    },
    "site": {
      "type": "string",
      "description": "Root folder of static assets for Workers Sites."
    },
    "siteInclude": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Array of .gitignore-style patterns that match file or directory names from the sites directory. Only matched items will be uploaded."
    },
    "siteExclude": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Array of .gitignore-style patterns that match file or directory names from the sites directory. Matched items will not be uploaded."
    },
    "var": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Array of key:value pairs to inject as variables into your code. The value will always be passed as a string to your Worker."
    },
    "define": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Array of key:value pairs to replace global identifiers in your code."
    },
    "triggers": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Cron schedules to attach to the deployed Worker. Refer to Cron Trigger Examples."
    },
    "routes": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Routes where this Worker will be deployed."
    },
    "tsconfig": {
      "type": "string",
      "description": "Path to a custom tsconfig.json file."
    },
    "minify": {
      "type": "boolean",
      "description": "Minify the bundled script before deploying."
    },
    "nodeCompat": {
      "type": "boolean",
      "description": "Enable node.js compatibility."
    },
    "keepVars": {
      "type": "boolean",
      "default": false,
      "description": "It is recommended best practice to treat your Wrangler developer environment as a source of truth for your Worker configuration, and avoid making changes via the Cloudflare dashboard. If you change your environment variables or bindings in the Cloudflare dashboard, Wrangler will override them the next time you deploy. If you want to disable this behavior set keepVars to true."
    }
  },
  "required": []
}
