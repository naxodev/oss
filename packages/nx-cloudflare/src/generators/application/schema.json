{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxCloudflareWorkerApplication",
  "title": "Create a Cloudflare Worker Application",
  "description": "Create a Cloudflare Worker Application",
  "type": "object",
  "properties": {
    "directory": {
      "description": "The directory of the new application.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Which directory do you want to create the application in?"
    },
    "name": {
      "description": "The name of the application.",
      "type": "string",
      "pattern": "^[a-zA-Z][^:]*$",
      "x-priority": "important"
    },
    "js": {
      "type": "boolean",
      "description": "Use JavaScript instead of TypeScript"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the application (used for linting)."
    },
    "frontendProject": {
      "type": "string",
      "description": "Frontend project that needs to access this application. This sets up proxy configuration.",
      "x-priority": "important"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "template": {
      "description": "Generate the initial worker using a template.",
      "type": "string",
      "enum": ["fetch-handler", "scheduled-handler", "hono", "none"],
      "default": "fetch-handler",
      "x-prompt": "Which worker template do you want to use?",
      "x-priority": "important"
    },
    "port": {
      "description": "The port in which the worker will be run on development mode",
      "type": "number",
      "default": 8787
    },
    "accountId": {
      "description": "The Cloudflare account identifier where the worker will be deployed",
      "type": "string"
    },
    "rootProject": {
      "description": "Create worker application at the root of the workspace",
      "type": "boolean",
      "default": false,
      "hidden": true,
      "x-priority": "internal"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["directory"]
}
