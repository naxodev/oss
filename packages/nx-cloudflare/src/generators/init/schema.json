{
  "$schema": "http://json-schema.org/schema",
  "$id": "Init",
  "title": "Init Nx Cloudflare Plugin",
  "type": "object",
  "properties": {
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "jest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "js": {
      "type": "boolean",
      "default": false,
      "description": "Use JavaScript instead of TypeScript"
    },
    "template": {
      "description": "Generate the initial worker using a template.",
      "type": "string",
      "enum": ["fetch-handler", "scheduled-handler", "hono", "none"]
    }
  },
  "required": ["name"]
}
