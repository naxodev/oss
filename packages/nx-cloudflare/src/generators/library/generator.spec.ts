import {
  getProjects,
  readJson,
  readProjectConfiguration,
  Tree,
  updateJson,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { NxCloudflareLibraryGeneratorSchema } from './schema';
import libraryGenerator from './generator';

describe('lib', () => {
  let tree: Tree;
  const defaultOptions: Omit<
    NxCloudflareLibraryGeneratorSchema,
    'name' | 'directory'
  > = {
    skipTsConfig: false,
    unitTestRunner: 'none',
    skipFormat: false,
    linter: 'eslint',
    js: false,
    strict: true,
    config: 'project',
  };

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    tree.write('/.gitignore', '');
    tree.write('/.gitignore', '');
  });

  describe('configs', () => {
    it('should generate an empty ts lib using --config=npm-scripts', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        config: 'npm-scripts',
      });
      expect(readJson(tree, '/my-lib/package.json')).toEqual({
        name: '@proj/my-lib',
        version: '0.0.1',
        private: true,
        scripts: {
          build: "echo 'implement build'",
          test: "echo 'implement test'",
        },
        dependencies: {},
      });

      expect(tree.exists('my-lib/src/index.ts')).toBeTruthy();
      expect(tree.exists('my-lib/src/lib/my-lib.ts')).toBeTruthy();

      // unitTestRunner property is ignored.
      // It only works with our executors.
      expect(tree.exists('my-lib/src/lib/my-lib.spec.ts')).toBeFalsy();
    });

    it('should generate an empty ts lib using --config=project', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        config: 'project',
      });
      const projectConfig = readProjectConfiguration(tree, 'my-lib');
      expect(projectConfig.root).toEqual('my-lib');
    });

    it('should generate an empty ts lib using --config=workspace', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        config: 'workspace',
      });
      const projectConfig = readProjectConfiguration(tree, 'my-lib');
      expect(projectConfig.root).toEqual('my-lib');
    });
  });

  describe('shared options', () => {
    describe('not-nested', () => {
      it('should update tags', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
          tags: 'one,two',
        });
        const projects = Object.fromEntries(getProjects(tree));
        expect(projects).toMatchObject({
          'my-lib': {
            tags: ['one', 'two'],
          },
        });
      });

      it('should work without directory parameter', async () => {
        // Test without specifying directory
        await libraryGenerator(tree, {
          ...defaultOptions,
          directory: 'my-lib-without-dir',
        });

        const project = readProjectConfiguration(tree, 'my-lib-without-dir');
        expect(project.root).toEqual('my-lib-without-dir');
        expect(tree.exists('my-lib-without-dir/src/index.ts')).toBeTruthy();
        expect(
          tree.exists('my-lib-without-dir/src/lib/my-lib-without-dir.ts')
        ).toBeTruthy();

        const tsconfigJson = readJson(tree, '/tsconfig.base.json');
        expect(
          tsconfigJson.compilerOptions.paths['@proj/my-lib-without-dir']
        ).toEqual(['my-lib-without-dir/src/index.ts']);
      });

      it('should update root tsconfig.base.json', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
        });
        const tsconfigJson = readJson(tree, '/tsconfig.base.json');
        expect(tsconfigJson.compilerOptions.paths['@proj/my-lib']).toEqual([
          'my-lib/src/index.ts',
        ]);
      });

      it('should update root tsconfig.json when no tsconfig.base.json', async () => {
        tree.rename('tsconfig.base.json', 'tsconfig.json');

        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
        });

        const tsconfigJson = readJson(tree, 'tsconfig.json');
        expect(tsconfigJson.compilerOptions.paths['@proj/my-lib']).toEqual([
          'my-lib/src/index.ts',
        ]);
      });

      it('should update root tsconfig.base.json (no existing path mappings)', async () => {
        updateJson(tree, 'tsconfig.base.json', (json) => {
          json.compilerOptions.paths = undefined;
          return json;
        });

        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
        });
        const tsconfigJson = readJson(tree, '/tsconfig.base.json');
        expect(tsconfigJson.compilerOptions.paths['@proj/my-lib']).toEqual([
          'my-lib/src/index.ts',
        ]);
      });

      it('should create a local tsconfig.json', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
        });
        const tsconfigJson = readJson(tree, 'my-lib/tsconfig.json');
        expect(tsconfigJson).toMatchInlineSnapshot(`
          {
            "compilerOptions": {
              "forceConsistentCasingInFileNames": true,
              "importHelpers": true,
              "module": "commonjs",
              "noFallthroughCasesInSwitch": true,
              "noImplicitOverride": true,
              "noImplicitReturns": true,
              "noPropertyAccessFromIndexSignature": true,
              "strict": true,
            },
            "extends": "../tsconfig.base.json",
            "files": [],
            "include": [],
            "references": [
              {
                "path": "./tsconfig.lib.json",
              },
            ],
          }
        `);
      });

      it('should extend from root tsconfig.json when no tsconfig.base.json', async () => {
        tree.rename('tsconfig.base.json', 'tsconfig.json');

        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
        });

        const tsconfigJson = readJson(tree, 'my-lib/tsconfig.json');
        expect(tsconfigJson.extends).toBe('../tsconfig.json');
      });
    });

    describe('nested', () => {
      it('should update tags', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
          tags: 'one',
        });
        let projects = Object.fromEntries(getProjects(tree));
        expect(projects).toMatchObject({
          'my-lib': {
            tags: ['one'],
          },
        });

        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib2',
          directory: 'my-dir/my-lib-2',
          tags: 'one,two',
        });
        projects = Object.fromEntries(getProjects(tree));
        expect(projects).toMatchObject({
          'my-lib': {
            tags: ['one'],
          },
          'my-lib2': {
            tags: ['one', 'two'],
          },
        });
      });

      it('should generate files', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
        });
        expect(tree.exists('my-dir/my-lib/src/index.ts')).toBeTruthy();
        expect(tree.exists('my-dir/my-lib/src/lib/my-lib.ts')).toBeTruthy();
        expect(tree.exists('my-dir/my-lib/src/index.ts')).toBeTruthy();
        expect(tree.exists(`my-dir/my-lib/.eslintrc.json`)).toBeTruthy();
        expect(tree.exists(`my-dir/my-lib/package.json`)).toBeTruthy();
      });

      it('should update project configuration', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
          config: 'workspace',
        });

        expect(readProjectConfiguration(tree, 'my-lib').root).toEqual(
          'my-dir/my-lib'
        );
      });

      it('should update root tsconfig.base.json', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
        });
        const tsconfigJson = readJson(tree, '/tsconfig.base.json');
        expect(tsconfigJson.compilerOptions.paths['@proj/my-lib']).toEqual([
          'my-dir/my-lib/src/index.ts',
        ]);
        expect(tsconfigJson.compilerOptions.paths['my-lib/*']).toBeUndefined();
      });

      it('should update root tsconfig.json when no tsconfig.base.json', async () => {
        tree.rename('tsconfig.base.json', 'tsconfig.json');

        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
        });

        const tsconfigJson = readJson(tree, '/tsconfig.json');
        expect(tsconfigJson.compilerOptions.paths['@proj/my-lib']).toEqual([
          'my-dir/my-lib/src/index.ts',
        ]);
        expect(tsconfigJson.compilerOptions.paths['my-lib/*']).toBeUndefined();
      });

      it('should create a local tsconfig.json', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
        });

        const tsconfigJson = readJson(tree, 'my-dir/my-lib/tsconfig.json');
        expect(tsconfigJson.references).toEqual([
          {
            path: './tsconfig.lib.json',
          },
        ]);
      });

      it('should extend from root tsconfig.base.json', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
        });

        const tsconfigJson = readJson(tree, 'my-dir/my-lib/tsconfig.json');
        expect(tsconfigJson.extends).toBe('../../tsconfig.base.json');
      });

      it('should extend from root tsconfig.json when no tsconfig.base.json', async () => {
        tree.rename('tsconfig.base.json', 'tsconfig.json');

        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
        });

        const tsconfigJson = readJson(tree, 'my-dir/my-lib/tsconfig.json');
        expect(tsconfigJson.extends).toBe('../../tsconfig.json');
      });
    });

    describe('--no-strict', () => {
      it('should update the projects tsconfig with strict false', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
          strict: false,
        });
        const tsconfigJson = readJson(tree, '/my-lib/tsconfig.json');

        expect(
          tsconfigJson.compilerOptions?.forceConsistentCasingInFileNames
        ).not.toBeDefined();
        expect(tsconfigJson.compilerOptions?.strict).not.toBeDefined();
        expect(
          tsconfigJson.compilerOptions?.noImplicitOverride
        ).not.toBeDefined();
        expect(
          tsconfigJson.compilerOptions?.noPropertyAccessFromIndexSignature
        ).not.toBeDefined();
        expect(
          tsconfigJson.compilerOptions?.noImplicitReturns
        ).not.toBeDefined();
        expect(
          tsconfigJson.compilerOptions?.noFallthroughCasesInSwitch
        ).not.toBeDefined();
      });

      it('should default to strict true', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
        });
        const tsconfigJson = readJson(tree, '/my-lib/tsconfig.json');

        expect(tsconfigJson.compilerOptions.strict).toBeTruthy();
        expect(
          tsconfigJson.compilerOptions.forceConsistentCasingInFileNames
        ).toBeTruthy();
        expect(tsconfigJson.compilerOptions.noImplicitReturns).toBeTruthy();
        expect(
          tsconfigJson.compilerOptions.noFallthroughCasesInSwitch
        ).toBeTruthy();
      });
    });

    describe('--importPath', () => {
      it('should update the tsconfig with the given import path', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
          importPath: '@myorg/lib',
        });
        const tsconfigJson = readJson(tree, '/tsconfig.base.json');

        expect(tsconfigJson.compilerOptions.paths['@myorg/lib']).toBeDefined();
      });

      it('should fail if the same importPath has already been used', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'myLib1',
          directory: 'myLib1',
          importPath: '@myorg/lib',
        });

        try {
          await libraryGenerator(tree, {
            ...defaultOptions,
            name: 'myLib2',
            directory: 'myLib2',
            importPath: '@myorg/lib',
          });
        } catch (e) {
          expect(e.message).toContain(
            'You already have a library using the import path'
          );
        }

        expect.assertions(1);
      });

      it('should provide a default import path using npm scope', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
        });

        const tsconfigJson = readJson(tree, '/tsconfig.base.json');
        expect(
          tsconfigJson.compilerOptions.paths['@proj/my-lib']
        ).toBeDefined();
      });

      it('should read import path from existing name in package.json', async () => {
        updateJson(tree, 'package.json', (json) => {
          json.name = '@acme/core';
          return json;
        });
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: '.',
        });

        const tsconfigJson = readJson(tree, '/tsconfig.base.json');
        expect(tsconfigJson.compilerOptions.paths['@acme/core']).toBeDefined();
      });
    });
  });

  describe('--linter', () => {
    it('should add eslint dependencies', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
      });

      const packageJson = readJson(tree, 'package.json');
      expect(packageJson.devDependencies['eslint']).toBeDefined();
      expect(packageJson.devDependencies['@nx/eslint']).toBeDefined();
      expect(packageJson.devDependencies['@nx/eslint-plugin']).toBeDefined();
    });

    describe('not nested', () => {
      it('should create a local .eslintrc.json', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-lib',
        });

        const eslintJson = readJson(tree, 'my-lib/.eslintrc.json');
        expect(eslintJson).toMatchInlineSnapshot(`
          {
            "extends": [
              "../.eslintrc.json",
            ],
            "ignorePatterns": [
              "!**/*",
            ],
            "overrides": [
              {
                "files": [
                  "*.ts",
                  "*.tsx",
                  "*.js",
                  "*.jsx",
                ],
                "rules": {},
              },
              {
                "files": [
                  "*.ts",
                  "*.tsx",
                ],
                "rules": {},
              },
              {
                "files": [
                  "*.js",
                  "*.jsx",
                ],
                "rules": {},
              },
              {
                "files": [
                  "*.json",
                ],
                "parser": "jsonc-eslint-parser",
                "rules": {
                  "@nx/dependency-checks": [
                    "error",
                    {
                      "ignoredFiles": [
                        "{projectRoot}/eslint.config.{js,cjs,mjs,ts,cts,mts}",
                      ],
                    },
                  ],
                },
              },
            ],
          }
        `);
      });
    });

    describe('nested', () => {
      it('should create a local .eslintrc.json', async () => {
        await libraryGenerator(tree, {
          ...defaultOptions,
          name: 'my-lib',
          directory: 'my-dir/my-lib',
        });

        const eslintJson = readJson(tree, 'my-dir/my-lib/.eslintrc.json');
        expect(eslintJson).toMatchInlineSnapshot(`
          {
            "extends": [
              "../../.eslintrc.json",
            ],
            "ignorePatterns": [
              "!**/*",
            ],
            "overrides": [
              {
                "files": [
                  "*.ts",
                  "*.tsx",
                  "*.js",
                  "*.jsx",
                ],
                "rules": {},
              },
              {
                "files": [
                  "*.ts",
                  "*.tsx",
                ],
                "rules": {},
              },
              {
                "files": [
                  "*.js",
                  "*.jsx",
                ],
                "rules": {},
              },
              {
                "files": [
                  "*.json",
                ],
                "parser": "jsonc-eslint-parser",
                "rules": {
                  "@nx/dependency-checks": [
                    "error",
                    {
                      "ignoredFiles": [
                        "{projectRoot}/eslint.config.{js,cjs,mjs,ts,cts,mts}",
                      ],
                    },
                  ],
                },
              },
            ],
          }
        `);
      });
    });
  });

  describe('--bundler=vite', () => {
    // extend timeout of test
    jest.setTimeout(100000);
    it('should add build and test targets with vite and vitest', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        bundler: 'vite',
        unitTestRunner: 'vitest',
      });

      expect(tree.exists('my-lib/vite.config.ts')).toBeTruthy();
      expect(tree.read('my-lib/vite.config.ts', 'utf-8')).toMatchSnapshot();
      expect(tree.read('my-lib/README.md', 'utf-8')).toMatchSnapshot();
      expect(tree.read('my-lib/tsconfig.lib.json', 'utf-8')).toMatchSnapshot();
      expect(tree.read('my-lib/.eslintrc.json', 'utf-8')).toMatchSnapshot();
    });
  });

  describe('--bundler=esbuild', () => {
    it('should add build with esbuild', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        bundler: 'esbuild',
        unitTestRunner: 'none',
      });

      const project = readProjectConfiguration(tree, 'my-lib');
      expect(project.targets.build).toMatchObject({
        executor: '@nx/esbuild:esbuild',
      });
      expect(tree.read('my-lib/.eslintrc.json', 'utf-8')).toMatchSnapshot();
    });
  });

  describe('--minimal', () => {
    it('should generate a README.md when minimal is set to false', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        minimal: false,
      });

      expect(tree.exists('my-lib/README.md')).toBeTruthy();
    });

    it('should not generate a README.md when minimal is set to true', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        minimal: true,
      });

      expect(tree.exists('my-lib/README.md')).toBeFalsy();
    });

    it('should generate a README.md and add it to the build assets when buildable is true and minimal is false', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        bundler: 'tsc',
        minimal: false,
      });

      expect(tree.exists('my-lib/README.md')).toBeTruthy();

      const project = readProjectConfiguration(tree, 'my-lib');
      expect(project.targets.build.options.assets).toStrictEqual([
        'my-lib/*.md',
      ]);
    });

    it('should not generate a README.md when both bundler and minimal are set', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        directory: 'my-lib',
        bundler: 'tsc',
        minimal: true,
      });

      expect(tree.exists('my-lib/README.md')).toBeFalsy();

      const project = readProjectConfiguration(tree, 'my-lib');
      expect(project.targets.build.options.assets).toEqual([]);
    });
  });

  describe('--simpleName', () => {
    it('should generate a simple name', async () => {
      await libraryGenerator(tree, {
        ...defaultOptions,
        name: 'my-lib',
        simpleName: true,
        directory: 'web/my-lib',
      });

      expect(tree.read('web/my-lib/src/index.ts', 'utf-8')).toContain(
        `export * from './lib/my-lib';`
      );
      expect(tree.exists('web/my-lib/src/lib/my-lib.ts')).toBeTruthy();
    });
  });
});
