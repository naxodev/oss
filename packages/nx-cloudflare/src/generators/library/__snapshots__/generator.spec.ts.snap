// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib --bundler=esbuild should add build with esbuild 1`] = `
"{
  "extends": ["../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/dependency-checks": [
          "error",
          {
            "ignoredFiles": [
              "{projectRoot}/eslint.config.{js,cjs,mjs}",
              "{projectRoot}/esbuild.config.{js,ts,mjs,mts}"
            ]
          }
        ]
      }
    }
  ]
}
"
`;

exports[`lib --bundler=vite should add build and test targets with vite and vitest 1`] = `
"/// <reference types='vitest' />
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';
import * as path from 'path';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { nxCopyAssetsPlugin } from '@nx/vite/plugins/nx-copy-assets.plugin';

export default defineConfig({
  root: __dirname,
  cacheDir: '../node_modules/.vite/my-lib',
  plugins: [
    nxViteTsPaths(),
    nxCopyAssetsPlugin(['*.md']),
    dts({
      entryRoot: 'src',
      tsconfigPath: path.join(__dirname, 'tsconfig.lib.json'),
    }),
  ],
  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },
  // Configuration for building your library.
  // See: https://vitejs.dev/guide/build.html#library-mode
  build: {
    outDir: '../dist/my-lib',
    emptyOutDir: true,
    reportCompressedSize: true,
    commonjsOptions: {
      transformMixedEsModules: true,
    },
    lib: {
      // Could also be a dictionary or array of multiple entry points.
      entry: 'src/index.ts',
      name: 'my-lib',
      fileName: 'index',
      // Change this to the formats you want to support.
      // Don't forget to update your package.json as well.
      formats: ['es'],
    },
    rollupOptions: {
      // External packages that should not be bundled into your library.
      external: [],
    },
  },
  test: {
    watch: false,
    globals: true,
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    reporters: ['default'],
    coverage: {
      reportsDirectory: '../coverage/my-lib',
      provider: 'v8',
    },
  },
});
"
`;

exports[`lib --bundler=vite should add build and test targets with vite and vitest 2`] = `
"# my-lib

This library was generated with [Nx](https://nx.dev).

## Building

Run \`nx build my-lib\` to build the library.

## Running unit tests

Run \`nx test my-lib\` to execute the unit tests via [Vitest](https://vitest.dev/).
"
`;

exports[`lib --bundler=vite should add build and test targets with vite and vitest 3`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/out-tsc",
    "declaration": true,
    "types": [
      "node",
      "vite/client",
      "@cloudflare/workers-types/experimental",
      "@cloudflare/vitest-pool-workers"
    ]
  },
  "include": ["src/**/*.ts"],
  "exclude": [
    "vite.config.ts",
    "vite.config.mts",
    "vitest.config.ts",
    "vitest.config.mts",
    "src/**/*.test.ts",
    "src/**/*.spec.ts",
    "src/**/*.test.tsx",
    "src/**/*.spec.tsx",
    "src/**/*.test.js",
    "src/**/*.spec.js",
    "src/**/*.test.jsx",
    "src/**/*.spec.jsx"
  ]
}
"
`;

exports[`lib --bundler=vite should add build and test targets with vite and vitest 4`] = `
"{
  "extends": ["../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/dependency-checks": [
          "error",
          {
            "ignoredFiles": [
              "{projectRoot}/eslint.config.{js,cjs,mjs}",
              "{projectRoot}/vite.config.{js,ts,mjs,mts}"
            ]
          }
        ]
      }
    }
  ]
}
"
`;
